packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -14,11 +14,24 @@ let buildWrapper = template(`
   })
 `);
 
+let arrowBuildWrapper =  template(`
+  (() => {
+    var ref = FUNCTION;
+    return (PARAMS) => ref.apply(this, arguments);
```

Comment:
Test `deeply-nested-asyncs` but if you suggests other tricky ideas of test case cases with async arrows, I'll add it and check. And of course everything requires validation.